server/aesdsocket.c

```diff
@@ -72,68 +102,159 @@ int main(int argc, char **argv)
     struct addrinfo hints;
     struct addrinfo *pServerInfo;
     int serverfd = -1;
+    int status;
 
     // Create logger
     openlog(NULL, 0, LOG_USER);
 
+    // Create storage file
+    status = create_storage_file();
+    if (status != 0)
+    {
+        log_message(LOG_ERR, "Error: creating storeage file errno=%d\n", status);
+        goto error;
+    }
+
     // Clear data structure
     memset(&hints, 0, sizeof(hints));
 
     // Get server info for IP address
     hints.ai_family = FAMILY;
     hints.ai_socktype = SOCKET_TYPE;
-    int status = getaddrinfo(NULL, PORT, &hints, &pServerInfo);
+    status = getaddrinfo(NULL, PORT, &hints, &pServerInfo);
     if (status != 0)
     {
         log_message(LOG_ERR, "Error: getaddrinfo() %s\n", gai_strerror(status));
-        goto socketerror;
+        goto error;
     }
 
     // Open socket connection
     serverfd = socket(FAMILY, SOCKET_TYPE, 0);
     if (serverfd < 0)
     {
         log_message(LOG_ERR, "Error: opening socket, errno=%d\n", errno);
-        goto socketerror;
+        goto error;
     }
 
     // Bind device address to socket
     if (bind(serverfd, pServerInfo->ai_addr, pServerInfo->ai_addrlen) < 0)
     {
         log_message(LOG_ERR, "Error: binding socket errno=%d\n", errno);
-        goto socketerror;
+        goto error;
     }
 
     // Listen for connection
     if (listen(serverfd, MAX_CONNECTIONS < 0))
     {
         log_message(LOG_ERR, "Error: listening for connection errno=%d\n", errno);
-        goto socketerror;
+        goto error;
     }
 
     log_message(LOG_INFO, "Listening for clients on port %s ...", PORT);
 
     // Listen for connection forever
     while (1)
     {
-        struct sockaddr_in clientAddr;            // Specified address of accepted client
+        struct sockaddr_in clientAddr;                 // Specified address of accepted client
         socklen_t clientAddrSize = sizeof(clientAddr); // Size of client address
         int clientfd;
+        char buf[1024];
+        ssize_t nRead;
+        ssize_t nWrite;
+        ssize_t spaceRemaining = sizeof(buf);
+        int streamPos = 0;
 
-        // Accept incoming connections.  This function will block until connection
+        // Accept incoming connections.  accept() will block until connection
         clientfd = accept(serverfd, (struct sockaddr *)&clientAddr, &clientAddrSize);
-        if(clientfd < 0)
+        if (clientfd < 0)
         {
             log_message(LOG_ERR, "Error: failed to accept client errno=%d\n", errno);
-            goto socketerror;
+            goto error;
         }
 
         log_message(LOG_INFO, "Accepted connection from %s", inet_ntoa(clientAddr.sin_addr));
-         
-        // Close connection with client 
+
+        // Read data from socket
+        while (1)
+        {
+            nRead = read(clientfd, &buf[streamPos], spaceRemaining);
```

Comment:
what if the data received  is more than 1024 bytes , you need to  use dynamic memory allocation , otherwise it wont work for assignment 8